src/schema/rules/dataset_metadata.yaml

```
@@ -0,0 +1,53 @@
+dataset_description:
+    selectors:
+    - path == "/dataset_description.json"
+    fields:
+        Name: REQUIRED
+        BIDSVersion: REQUIRED
+        DatasetType: RECOMMENDED
+        License: RECOMMENDED
+        Authors: RECOMMENDED
+        Acknowledgements: OPTIONAL
+        HowToAcknowledge: OPTIONAL
+        Funding: OPTIONAL
+        EthicsApprovals: OPTIONAL
+        ReferencesAndLinks: OPTIONAL
+        DatasetDOI: OPTIONAL
+        GeneratedBy: RECOMMENDED
+        GeneratedBy[].Name: REQUIRED
+        GeneratedBy[].Version: RECOMMENDED
+        GeneratedBy[].Description: OPTIONAL
+        GeneratedBy[].CodeURL: OPTIONAL
+        GeneratedBy[].Container: OPTIONAL
+        GeneratedBy[].Container.Type: RECOMMENDED
+        GeneratedBy[].Container.Tag: RECOMMENDED
+        GeneratedBy[].Container.URI: RECOMMENDED
+        SourceDatasets: RECOMMENDED
+
+derivative_description:
+    selectors:
+    - path == "/dataset_description.json"
+    - json.DatasetType == "derivative"
+    fields:
+        GeneratedBy: REQUIRED
+
+dataset_description_with_genetics:
+    selectors:
+    - path == "/dataset_description.json"
+    - dataset.files contains "/genetic_info.json"
+    fields:
+        Genetics: REQUIRED
+        Genetics.Dataset: REQUIRED
+        Genetics.Database: OPTIONAL
+        Genetics.Descriptors: OPTIONAL
```

@tsalo Would this correspond to these names in the metadata.yaml as well?

```suggestion
        Genetics{}.Dataset: REQUIRED
        Genetics{}.Database: OPTIONAL
        Genetics{}.Descriptors: OPTIONAL
```

---

Well, I want to move those names _into_ the `Genetics` object in metadata.yaml anyway, so I'm not too concerned about `Genetics.Dataset`, etc., in that file. We just need a new function to make tables of subobjects for that to work, since there's a table with the Genetics subobjects in the specification.

One random thought though- do we need the dot after the braces?